LibTCPServer.emx

```diff
@@ -670,65 +663,83 @@ str.append("}");
 if (config.logToStdOut)
 	std::cout << "Out message: " << str << std::endl;
 
-bool writeSuccess = false;
-try {
-	SocketAddress socketAddress(config.remoteHost, config.remotePort);
-	StreamSocket socket(socketAddress);
-	socket.sendBytes(str.c_str(), str.length());
+	bool writeSuccess = false;
 
+	RTTcpSocket tcp;
+	RTIOMonitor ioMonitor;
+	tcp.registerWith(&ioMonitor);
+	if (!tcp.create()) {
+		std::cerr << "TCP socket creation is failed." << std::endl;
+		tcp.close();
+		return;
+	}
+	if (!tcp.connect(static_cast<const char*>(config.remoteHost.c_str()),
+			static_cast<int>(config.remotePort))) {
+		std::cerr << "Unable to connect to " << config.remoteHost << " on port "
+				<< config.remotePort << std::endl;
+		tcp.close();
+		return;
+	}
+	if (tcp.write(static_cast<const char*>(str.c_str()),
+			static_cast<int>(str.length())) < 0) {
+		std::cerr << "Error: Failed to write to socket for "
+				<< config.remoteHost << " on port " << config.remotePort
+				<< std::endl;
+		tcp.close();
+		return;
+	}
 	writeSuccess = true;
 
+	static const RTTimespec awhile(1L, 0L);
+	while (!ioMonitor.wait(&awhile)) {
+		std::this_thread::sleep_for(std::chrono::seconds(1));
```

Comment:
What do we gain by two waits? And what will happen if no response at all comes for the request? Will we get stuck in an infinite loop?

Reply:
I used maxWaitForReply configuration and return if response is not coming within time.